src/main/java/de/unistuttgart/towercrushbackend/controller/websockets/WebsocketController.java

```diff
@@ -12,73 +12,34 @@
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
 
 @Controller
 @Slf4j
 public class WebsocketController {
 
     @Autowired
     SimpMessagingTemplate simpMessagingTemplate;
-    String destination = "/topic/messages";
-    String gameDestination = "/topic/games/";
-
-    ExecutorService executorService =
-        Executors.newFixedThreadPool(3);
-    Map<String, Future<?>> submittedLobbies = new HashMap<>();
-    @MessageMapping("/start/lobbyName/{lobbyName}")
-    public void startGame(@DestinationVariable String lobbyName) {
-        log.info("Started task");
-        MessageWrapper startMessage = new MessageWrapper("Started game: " + lobbyName, Purpose.CHAT_MESSAGE);
-        simpMessagingTemplate.convertAndSend(gameDestination + lobbyName,
-            startMessage);
-        submittedLobbies.put(lobbyName, executorService.submit(() -> {
-            while (true) {
-                MessageWrapper idleMessage = new MessageWrapper(LocalDateTime.now()
-                    + ": doing game: " + lobbyName, Purpose.CHAT_MESSAGE);
-                simpMessagingTemplate.convertAndSend(gameDestination + lobbyName,
-                    idleMessage);
-                Thread.sleep(10000);
-            }
-        }));
+    final String destination = "/topic/messages";
+    final String lobbyDestination = "/topic/lobbies/";
+
+    @MessageMapping("/lobby/{lobby}/team/{team}/player/{player}")
+    public void joinTeam(@DestinationVariable final String lobby, @DestinationVariable final String team, @DestinationVariable final String player) throws JsonProcessingException {
+        log.info("player: " + player + ", joined team: " + team + ", in lobby: " + lobby);
+        final JoinTeamMessage joinTeamMessage = new JoinTeamMessage(team, player);
+        final String jsonString = convertObjectToJson(joinTeamMessage);
+        final MessageWrapper joinTeamMessageWrapped = new MessageWrapper(jsonString, Purpose.JOIN_TEAM_MESSAGE);
+        simpMessagingTemplate.convertAndSend(lobbyDestination + lobby, joinTeamMessageWrapped);
     }
 
-    @MessageMapping("/lobbyName/{lobbyName}/team/{teamName}/player/{playerName}")
-    public void joinTeam(@DestinationVariable String lobbyName, @DestinationVariable String teamName, @DestinationVariable String playerName) throws JsonProcessingException {
-        log.info("join team");
-        JoinTeamMessage joinTeamMessage = new JoinTeamMessage(teamName,playerName);
-        String jsonString = convertObjectToJson(joinTeamMessage);
-        MessageWrapper joinTeamMessageWrapped = new MessageWrapper(jsonString, Purpose.JOIN_TEAM_MESSAGE);
-        simpMessagingTemplate.convertAndSend(gameDestination + lobbyName,
-            joinTeamMessageWrapped);
+    @MessageMapping("/start/lobby/{lobby}")
+    public void startLobby(@DestinationVariable final String lobby) {
+        log.info("start lobby");
```

Comment:
also needs to be adapted, according to the answer above. Should also contain the lobby parameter.

Reply:
i dont understand this comment

Reply:
He means if you change startLobby method to startGame you should also log "start game" here instead of "start lobby".
Also log the lobby parameter that is used in this method so you know which lobby gets started.

Reply:
ah yea well, this method will be implemented later there is also a todo inside so i mean this is completly irrelevant no?

Reply:
Somewhat. If you adapt it then yes.